src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -496,7 +499,7 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 		}
 
 		for _, dp := range value.Datapoints {
-			if value.Attributes.PromType != ts.PromMetricTypeUnknown {
+			if d.storeMetricType && value.Attributes.PromType != ts.PromMetricTypeUnknown {
 				switch value.Attributes.PromType {
 				case ts.PromMetricTypeCounter:
 					err = result.SamplesAppender.AppendCounterTimedSample(dp.Timestamp, int64(dp.Value))
```

Comment:
Hm I'm a little concerned about sending Prometheus counters to the aggregator with `AppendCounterTimedSample` since Prometheus counters can actually be in increments that aren't int64 FYI.

i.e. it's totally valid to increment a counter by 0.5 in Prometheus. Notice the API here:
https://godoc.org/github.com/prometheus/client_golang/prometheus#Counter
```go
type Counter interface {
    Metric
    Collector

    // Inc increments the counter by 1. Use Add to increment it by arbitrary
    // non-negative values.
    Inc()
    // Add adds the given value to the counter. It panics if the value is <
    // 0.
    Add(float64)
}
```

Maybe out of scope of this PR, but we'll have to think about how to pass the metric type to the aggregator without using the AppendCounterTimedSample (and continue to use AppendGaugeTimedSample) I think somehow in the future.